src/bundle/Controller/ContentOnTheFlyController.php

```diff
@@ -253,17 +287,31 @@ public function editContentAction(
         }
 
         if ($form->isSubmitted() && $form->isValid() && null !== $form->getClickedButton()) {
-            $this->createContentActionDispatcher->dispatchFormAction(
+            $actionName = $form->getClickedButton()->getName();
+
+            $actionDispatcher = $actionName === 'autosave'
+                ? $this->contentActionDispatcher
+                : $this->createContentActionDispatcher;
+
+            if (empty($location)) {
+                $contentInfo = $this->contentService->loadContentInfo($content->id);
+
+                if (!empty($contentInfo->mainLocationId)) {
+                    $location = $this->locationService->loadLocation($contentInfo->mainLocationId);
+                }
+            }
+
+            $actionDispatcher->dispatchFormAction(
                 $form,
                 $form->getData(),
                 $form->getClickedButton()->getName(),
                 ['referrerLocation' => $location]
             );
 
-            if ($this->createContentActionDispatcher->getResponse()) {
+            if ($actionDispatcher->getResponse()) {
                 $view = new EditContentOnTheFlySuccessView('@ezdesign/ui/on_the_fly/content_edit_response.html.twig');
                 $view->addParameters([
-                    'locationId' => $location->id,
+                    'locationId' => empty($location) ? null : $location->id,
```

Comment:
Can we use  `$location instanceof Location` or `$location !== null` instead of `empty($location)` ?